src/accelerate/test_utils/testing.py

```diff
@@ -193,6 +211,39 @@ def require_non_xpu(test_case):
     return unittest.skipUnless(torch_device != "xpu", "test requires a non-XPU")(test_case)
 
 
+def require_non_hpu(test_case):
+    """
+    Decorator marking a test that should be skipped for HPU.
+    """
+    return unittest.skipUnless(torch_device != "hpu", "test requires a non-HPU")(test_case)
+
+
+def require_fp16(test_case):
+    """
+    Decorator marking a test that requires FP16. These tests are skipped when FP16 is not supported.
+    """
+
+    return unittest.skipUnless(is_fp16_available(), "test requires FP16 support")(test_case)
+
+
+def require_fp8(test_case):
+    """
+    Decorator marking a test that requires FP8. These tests are skipped when FP8 is not supported.
+    """
+
+    # is_fp8_available only checks for libraries
+    # ideally it should check for device capability as well
+    fp8_is_available = is_fp8_available()
```

Comment:
it seems that `is_fp8_available()` only checks for libraries availability and not for device capability.
`check_fp8_capability` is used for that but that's confusing no ? because is_fp16/bf16 do check for device capability of using those dtypes.

Reply:
I didn't change the behavior of `is_fp8_available()` to not break bc but it would make sense to have a single source of truth.

Reply:
I can see the logic, I'll change it in a follow up and can see why that'd be confusing